### PR TITLE
refactor(es): rework TLS configuration for Elasticsearch

### LINKED ISSUE #1195: Elasticsearch basic auth credentials are ignored when using https

### PR DESCRIPTION
**Pull request by volkflo**
Closes #1195

### CODE REVIEW

File: common/es/src/main/java/io/apiman/common/es/util/DefaultEsClientFactory.java [hunk 1]
```diff
@@ -198,12 +195,13 @@ private void waitForElasticsearch(RestHighLevelClient client, long pollingTime)
...
                             final Date endTime = new Date();
                             long pollingTimeMeasure = endTime.getTime() - startTime.getTime();
-                            logger.info("Took "+ pollingTimeMeasure + " milliseconds for polling Elasticsearch");
-
+                            logger.info("Took " + pollingTimeMeasure + " milliseconds for polling Elasticsearch");
                             // wake up the waiting thread
                             cdl.countDown();
                         }
-
+                    } catch (ElasticsearchException | SSLException e) {
```
Comment (msavy) [inline common/es/src/main/java/io/apiman/common/es/util/DefaultEsClientFactory.java:202]: Can you try just throwing these and see if it works?

And if not, I can tell you a few ways to handle this properly. For example, the holder can say whether it's a fatal error or not.

File: common/es/src/main/java/io/apiman/common/es/util/DefaultEsClientFactory.java [hunk 2]
```diff
@@ -198,12 +195,13 @@ private void waitForElasticsearch(RestHighLevelClient client, long pollingTime)
...
+                        logger.error("Error connecting to Elasticsearch: ", e);
+                        exception.setValue(e);
                     } catch (IOException e) {
                         logger.info("Unable to reach Elasticsearch. Will continue polling.");
```
Comment (msavy) [inline common/es/src/main/java/io/apiman/common/es/util/DefaultEsClientFactory.java:206]: In the logger let's at least print out the exception message to guide us, let's say:

```
logger.info("Unable to reach Elasticsearch with error ({0}). Will continue polling.", e.getMessage())
```

Comment (volkflo) [reply common/es/src/main/java/io/apiman/common/es/util/DefaultEsClientFactory.java:206]: Will test everything, was just a draft pr to test if CI fails :)

**Comment by volkflo** [comment]
I think this good to go for the moment.

- TLS is working now :heavy_check_mark:
- IO Exception Logging without annoying stack trace :heavy_check_mark:
- ES/TLS Exceptions with stack trace e.g. to find the wrong password :heavy_check_mark:

@msavy I think we could improve this e.g. that we stop the loop if the password is wrong but I think that is not that important at the moment.

**Review by msavy — APPROVED** [review]
I think this should be good to go now. TLS config fixes look good to me. Do we have this reflected in the example/sample configs and docs?

I've pulled out the connection polling code into a separate simple class that manages the process, which makes it easier to understand and doesn't need any hacks because all the code was enclosed within a method previously.

The other PR I opened has a fix for the Elasticsearch warning spam, so let's not replicate it here.

**Comment by msavy** [comment]
BTW just in case you didn't notice, your commit message says 'Elasticse<b>ra</b>ch' instead of Elasticsearch 😅

**Pull request merged**